From c9c5d3d9c90ab9c8dcdb991ddff97866fbe59c99 Mon Sep 17 00:00:00 2001
From: Sveinung Kvilhaugsvik <sveinung84@users.sourceforge.net>
Date: Sat, 31 Mar 2018 20:47:00 +0200
Subject: [PATCH] AI: Fix city targeted spy actions.

Stop confusing the target city with the target extra.

See hrm Bug #743594
---
 ai/default/aidiplomat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ai/default/aidiplomat.c b/ai/default/aidiplomat.c
index 2dbad1e7ac..888310807a 100644
--- a/ai/default/aidiplomat.c
+++ b/ai/default/aidiplomat.c
@@ -389,7 +389,7 @@ static void dai_diplomat_city(struct ai_type *ait, struct unit *punit,
              nation_rule_name(nation_of_unit(punit)),                      \
              unit_rule_name(punit), punit->id, city_name_get(ctarget));    \
     unit_do_action(pplayer, punit->id,                                     \
-                   EXTRA_NONE, ctarget->id, my_val, "", my_act);           \
+                   ctarget->id, EXTRA_NONE, my_val, "", my_act);           \
     return;                                                                \
   }
 
-- 
2.11.0

